tensorflow/src/TensorFlow/Types.hs

```diff
@@ -95,66 +100,31 @@ import Proto.Tensorflow.Core.Framework.Types (DataType(..))
 import TensorFlow.Internal.VarInt (getVarInt, putVarInt)
 import qualified TensorFlow.Internal.FFI as FFI
 
--- | Data about a tensor that is encoded for the TensorFlow APIs.
-newtype TensorData a = TensorData { unTensorData :: FFI.TensorData }
-
 -- | The class of scalar types supported by tensorflow.
-class TensorType a where
+class TensorDataType V.Vector a => TensorType a where
```

Comment:
It just reduced the code churn in our tests, but on second thought that isn't worth it.

I've made `TensorType` a superclass of `TensorDataType` instead, but that is just for convenience. Let me know what you think.